src/components/componentContainer.ts

```diff
@@ -19,7 +20,9 @@ export class ComponentContainer extends Component {
     this._detachCallback = (component: Component) => this.remove(component);
   }
 
-  public anchor(selection: d3.Selection<void>) {
+  public anchor(selection: SimpleSelection<void>) {
+    // coerce possibly external d3 instance into our own instance of d3 so we can use d3-selection-multi
```

Comment:
instead of copy+pasting this comment around, maybe we should just make a utils method for this? that might also help us track all the places where we do this sort of coercion.